replica/tablets.cc

```diff
@@ -156,6 +156,65 @@ future<> save_tablet_metadata(replica::database& db, const tablet_metadata& tm,
     co_await db.apply(freeze(muts), db::no_timeout);
 }
 
+static std::tuple<sstring, table_id> to_tablet_metadata_key(const schema& s, const partition_key& key) {
+    const auto elements = key.explode(s);
+    auto keyspace_name = value_cast<sstring>(utf8_type->deserialize_value(elements[0]));
+    auto table_id = value_cast<utils::UUID>(uuid_type->deserialize_value(elements[1]));
+    return std::tuple(std::move(keyspace_name), ::table_id(table_id));
+}
+
+static dht::token to_tablet_metadata_row_key(const schema& s, const clustering_key& key) {
+    const auto elements = key.explode(s);
+    return dht::token::from_int64(value_cast<int64_t>(long_type->deserialize_value(elements[0])));
+}
+
+static void do_update_tablet_metadata_change_hint(locator::tablet_metadata_change_hint& hint, const schema& s, const mutation& m) {
+    const auto [keyspace_name, table_id] = to_tablet_metadata_key(s, m.key());
+    auto it = hint.tables.try_emplace(table_id, locator::tablet_metadata_change_hint::table_hint{keyspace_name, table_id, {}}).first;
+
+    const auto& mp = m.partition();
+    if (mp.partition_tombstone() || !mp.row_tombstones().empty()) {
+        // If there is a partition tombstone or range tombstone, update the
+        // entire partition.
+        return;
+    }
+
```

Comment:
Need to also handle static column changes.

Reply:
Fixed in v5 (also added tests for this case).